File: 1481-least-number-of-unique-integers-after-k-removals/1481-least-number-of-unique-integers-after-k-removals.java
class Solution {
    public int findLeastNumOfUniqueInts(int[] arr, int k) {
        HashMap<Integer, Integer> mp = new HashMap<>(); ;
        for(Integer x : arr) mp.put(x , mp.getOrDefault(x , 0) + 1);
        
        List<Integer> vp = new ArrayList<>() ;
        for(Map.Entry<Integer,Integer> x : mp.entrySet() ){
            vp.add(x.getValue()) ;
        }
        
        Collections.sort(vp); 
        
        int count = 0;
        for (int i = 0 ; i < vp.size() ; i ++) {
            if (k >= vp.get(i)) {
                k -= vp.get(i);
                count++;
            } else {
                break;
            }
        }
        
        return vp.size() - count;
    }
}

File: 1481-least-number-of-unique-integers-after-k-removals/1481-least-number-of-unique-integers-after-k-removals.cpp
class Solution {
public:

    
    int findLeastNumOfUniqueInts(vector<int>& arr, int k) {
        unordered_map<int,int> mp ;
        for(auto x : arr) mp[x] ++;
        
        vector<int> vp ;
        for(auto &[l ,r] : mp){
            vp.push_back(r) ;
        }
        
        sort(vp.begin() , vp.end()) ;
        
        int count = 0;
        for (int& f : vp) {
            if (k >= f) {
                k -= f;
                count++;
            } else {
                break;
            }
        }
        
        return vp.size() - count;
    }
};